cli/command/stack/config.go

```diff
@@ -0,0 +1,62 @@
+package stack
+
+import (
+	"fmt"
+
+	"github.com/docker/cli/cli"
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/cli/cli/command/stack/loader"
+	"github.com/docker/cli/cli/command/stack/options"
+	composeLoader "github.com/docker/cli/cli/compose/loader"
+	composetypes "github.com/docker/cli/cli/compose/types"
+	"github.com/spf13/cobra"
+	yaml "gopkg.in/yaml.v2"
+)
+
+func newConfigCommand(dockerCli command.Cli) *cobra.Command {
+	var opts options.Config
+
+	cmd := &cobra.Command{
+		Use:     "config [OPTIONS]",
+		Aliases: []string{"cfg"},
+		Short:   "Outputs the final config file, after doing merges and interpolations",
+		Args:    cli.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			configDetails, err := loader.GetConfigDetails(opts.Composefiles, dockerCli.In())
```

Comment:
question: @SMFloris why not using `LoadComposefile` function instead, which actually calls `getConfigDetails` ?

Reply:
Great question!

If I recall correctly, I didn't use it because I wanted to have control on how the compose files are actually getting loaded.
The `--skip-interpolation` flag is being sent down to the `composeLoader.load()` call, like so:

```
optsFunc := func(options *composeLoader.Options) {
	options.SkipInterpolation = skipInterpolation
}
config, err := composeLoader.Load(configFiles, optsFunc)
```
This is kinda important to the use-case of this new command, seeing the output with and without interpolation to figure out where things went wrong and if you're deploying the right thing.

Another reason is that `LoadComposefile` also does some other things besides loading compose files, it also does some validation (forbidden, deprecated and unsupported properties). I though those validations were just for deploy since the validation of the `composeLoader.load` is also used. 🤔  Also, since in my mind I'm piping the output back to `docker stack deploy -c -` I saw no reason in doing deploy validations when not doing a deploy.

The final reason, since I wasn't that familiar with the code-base and because I added a new command I didn't want to mess too much with the existing signatures of the methods 😅 

Now that I'm more familiar with the codebase. I can change the `LoadComposefile` function to pass down some options to `composeLoader.load()` call if you want me too (perhaps even adding an option to skip those other validations).

Reply:
@thaJeztah any thoughts on that? I'm a bit on the fence, as if the output is pipped to the docker stack deploy command, then the config should print the warnings I guess... but it can also be pipped to the docker-compose command...
@SMFloris propose to call `LoadComposeFile` and add the `--skip-validation` option so one can disable the validation... WDYT?

Reply:
@thaJeztah , could we get your input on the above, please? 😄 We are doing some ugly hacks with downgrading docker-compose in order to get valid docker stack deploy files

Reply:
@thaJeztah , I managed to track the inclusion of these options down to https://github.com/docker/cli/pull/1128 

Maybe reading through that a bit can help jog some memories 😄